components/launcher/src/service.rs

```diff
@@ -89,8 +90,15 @@ fn pipe_stdout<T>(out: T, id: &str)
     let mut reader = BufReader::new(out);
     let mut buffer = String::new();
     while reader.read_line(&mut buffer).unwrap() > 0 {
-        let line = output_format!(preamble &id, logkey "O", buffer);
-        writeln!(&mut io::stdout(), "{}", line).expect("unable to write to stdout");
+        let so = StructuredOutput::new(&id,
+                                       "O",
+                                       line!(),
+                                       file!(),
+                                       column!(),
```

Comment:
Are all these line/file/column numbers going to pertain to _this_ code (i.e., always going to be the same)?

Is that expected / OK?

Reply:
They are and this PR does not change this but maybe we should change that here because it does seem rather useless and noisy. This just emits output from the process. What would you think of forcing this launcher output to always be non verbose which would remove these details?

Reply:
This is what happened before when we used the [`output_format!` macro](https://github.com/habitat-sh/core/blob/bba280adf5c46e57569caa1e15034747970ec245/components/core/src/output.rs#L264), right?

That said, I think we *should* fix this, and I think the most appropriate way is by making the `line`, `file` and `column` data associated specifically with the `Verbose` variant of the `Verbosity` enum.

Since we don't want to expose that to callers, we'll need a separate internal version
```rust
enum OutputVerbosityInternal {
    Normal,
    Verbose {
        line:   u32,
        file:   &'static str,
        column: u32,
    },
}
```
Then we simplify the struct:
```diff
 pub struct StructuredOutput<'a> {
     preamble: &'a str,
     logkey: &'static str,
-    line: u32,
-    file: &'static str,
-    column: u32,
     content: &'a str,
     /// The verbosity level of rendered content
-    verbosity: OutputVerbosity,
+    verbosity: OutputVerbosityInternal,
```
…add a bit of logic to `new`/`colored`:
```rust
        let verbosity = match verbosity {
            OutputVerbosity::Normal => OutputVerbosityInternal::Normal,
            OutputVerbosity::Verbose => OutputVerbosityInternal::Verbose { line, file, column },
        };
```
…change the verbosity `if let`s to:
```rust
                if let OutputVerbosityInternal::Verbose { line, file, column } = self.verbosity {
```
…and add a `succinct` associated function which creates the struct like:
```rust
        StructuredOutput { preamble,
                           logkey,
                           content,
                           verbosity: OutputVerbosityInternal::Normal,
                           format,
                           color_spec: ColorSpec::new() }
```
and the launcher can call:
```rust
        let so = StructuredOutput::succinct(&id, "O", output::get_format(), content);
```